sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -97,6 +97,12 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
       if (!Enqueued && EnqueueResultT::SyclEnqueueFailed == Res.MResult)
         throw runtime_error("Enqueue process failed.", PI_INVALID_OPERATION);
 
+      if (NewCmd->MDeps.size() == 0 && NewCmd->getQueue() != DefaultHostQueue) {
```

Comment:
It seems we need to follow up on host device then.

Reply:
not going that route. Changed.